src/gsi/gsi_rfv3io_mod.f90

```diff
@@ -1217,10 +1228,11 @@ subroutine read_fv3_netcdf_guess(fv3filenamegin)
         ntracerio2d=0
       endif
  
-      if( if_model_dbz )then
+      if( if_model_dbz .or. if_model_fed)then
```

Comment:
how about test " if (nphyvario >0)  then "

Reply:
@TingLei-NOAA 
I change the statement to: 
    if( allocated(fv3lam_io_phymetvars3d_nouv) )then
This is similar to how the traces, dynvars are treated in that code section.